src/main/java/org/broadinstitute/hellbender/tools/spark/sv/StructuralVariantContext.java

```diff
@@ -0,0 +1,280 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import htsjdk.samtools.*;
+import htsjdk.variant.variantcontext.StructuralVariantType;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFConstants;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.GATKSVVCFConstants;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.GATKSVVCFHeaderLines;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.haplotype.Haplotype;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * Variant context with additional method to mind the structural variant specific information from structural
+ * variant records.
+ */
+public class StructuralVariantContext extends VariantContext {
+
+    private static final long serialVersionUID = 1L;
+
+    private int end = -1;
+    private int length = -1;
+
+    /**
+     * Returns a instance given a plain {@link VariantContext} instance.
+     * <p>
+     *     This method will fail with a {@link IllegalArgumentException} if the input variant context is not
+     *     structural. Currently we only check that it has been properly annotated with the {@value VCFConstants#SVTYPE}
+     *     annotation.
+     * </p>
+     * @param vc the input variant context.
+     * @throws IllegalArgumentException if {@code vc} is {@code null} or does not seem to be
+     * a structural variant context based on its annotations.
+     * @return never {@code null}, potentially the same as the input if it happens to be an instance of this class.
+     */
+    public static StructuralVariantContext create(final VariantContext vc) {
+        if (vc instanceof StructuralVariantContext) {
+            return (StructuralVariantContext) vc;
+        } else {
+            checkIsStructuralVariantContext(vc);
+            return new StructuralVariantContext(vc);
+        }
+    }
+
+    private static void checkIsStructuralVariantContext(final VariantContext vc) {
+        Utils.nonNull(vc, "the input variant context must not be null");
+        Utils.nonNull(vc.getAttributeAsString(VCFConstants.SVTYPE, null), "the SVTYPE annotation is missing");
+    }
+
+    private StructuralVariantContext(final VariantContext other) {
+        super(other);
+    }
+
+    /**
```

Comment:
For individual breakends I guess the length is in fact 0 a breakpoint does not have dimensions. We can always add some ```if``` to handle this if needed.